### ISSUE TITLE
[prefer-literal-enum-member] option to allow bitwise operators

### ISSUE THREAD

**Issue by noftaly** [typescript-eslint/typescript-eslint#3487]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

*This is a suggestion*

**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/prefer-literal-enum-member": "error"
  }
}
```

```TS
enum Test {
    First = 1 << 0,
    Second = 1 << 1,
    Third = 1 << 2,
}
```

Bitwise in enums are very often used to act as flags that can overlap (there can be multiple of them).
*At least I used them often*

**Expected Result**

This is a suggestion to add the ability to use bitwise enums while having the `prefer-literal-enum-member` enabled. This can either work by default (which I don't think is the great approach as all the other expressions are banned) or behind a flag such as `allowBitwiseExpressions`.

**Actual Result**

There is a linting error telling me to use literal types.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.26.0` |
| `@typescript-eslint/parser`        | `4.26.0` |
| `TypeScript`                       | `4.3.2` |
| `ESLint`                           | `7.27.0` |
| `node`                             | `15.14.0` |
